iree/compiler/Dialect/HAL/IR/test/allocator_op_folding.mlir

```diff
@@ -0,0 +1,36 @@
+// RUN: iree-opt -allow-unregistered-dialect -split-input-file -canonicalize -cse %s | iree-opt -allow-unregistered-dialect -split-input-file | IreeFileCheck %s
```

Comment:
Is `-allow-unregistered-dialect` needed here? You remove it from other files in this PR. (maybe file an issue to triage those and remove, if it's not a state we want to stay in)